test/unit/assignments/functionPermutations.ts

```diff
@@ -133,6 +133,22 @@ export const selfTestFunctions: TestFunction[] = [
         }
         const anonFunctionNestedInNoSelfClass = (new AnonFunctionNestedInNoSelfClass).method();`,
     },
+    {
+        value: "anonMethodClassMergedNoSelfNS.method",
+        definition: `class AnonMethodClassMergedNoSelfNS { method(s: string): string { return s; } }
+            /** @noSelf */ namespace AnonMethodClassMergedNoSelfNS { export function nsFunc(s: string) { return s; } }
```

Comment:
Do these actually fail without the transformer change?

Reply:
Yes, because the class method will have it's `self` stripped due to the namespace using `@noSelf`

Reply:
But the call to the function will adjust accordingly?

Reply:
Yes, but the assignment test would fail since it would try to assign the method to a type with a `self` parameter. Also, those tests actually call the function and verify parameters are in correct order. That could probably be moved out to separate tests, but that seems out of scope for this PR.